### DISCUSSION TITLE
Wrap ConfirmDialog in Component

### DISCUSSION THREAD

**Discussion by bhaidar** [primefaces/community#442] [PrimeVue] [answered]
Hi,
I am trying to wrap the ConfirmDialog in a component to make it more reusable instead of having to re-configure the service every time I need it.

Here's a running example:
https://stackblitz.com/edit/kd8spn?file=src%2FConfirmButton.vue,src%2FApp.vue

However, it seems, I can't listen to the `accept` and `reject` events inside the ConfirmButton. Nothing is printed out when the dialog opens.

Is it possible to do this or I just forget about it?

Thank you
Bill

**Comment by tugcekucukoglu** [comment, marked as answer]
`accept` and `reject` are not valid emits for ConfirmDialog. Instead `onShow` and `onHide` Confirmation Options can be used.

**Reply by bhaidar** [reply]
What are the event details I receive as part of onHide and onShow. I need to know whether the confirmation was accepted or rejected. Thanks @tugcekucukoglu

**Comment by Rakasch** [comment]
Your example is basically working fine for me - on first load.
It stops working after the first confirmation.
Your problem is not the events that are emitted.
You need to call confirm.require each time the button is clicked. 
The ConfirmButton is only mounted once, and therefor your code is only working once.

**Reply by bhaidar** [reply]
Not really. I use `v-if` to show/hide. I assume, Vue would re-run `onMounted` when adding/removing the component from DOM.

I noticed now that reject/accept work fine. It's only when I use ESC to close the window, neither reject nor accept were executed.

I added `onHide` to emit the reject event and now works like a charm.

Check the update sample above.


Regards
Bill
